templates/aws-stack.yml

```diff
@@ -489,6 +490,11 @@ Parameters:
       - "false"
     Default: "false"
 
+  InstanceName:
+    Type: String
+    Description: Optional - Customise the EC2 instance Name tag
+    Default: "buildkite-agent"
```

Comment:
🤔 i wonder if we should abandon backwards compatibility here and have the default be 
```suggestion
    Default: !Ref 'AWS::StackName'
```

instead... given that instance names are non-unique already, they aren't going to be useful for addressing individual instances.

this would solve an issue that i have when i'm digging through the aws console sometimes where i look at this screen:
![CleanShot 2023-01-31 at 10 10 07@2x](https://user-images.githubusercontent.com/15753101/215595905-0bf4fc08-7b9b-4946-bcde-28ed91c59453.png)

and it's impossible to know which instance is from which elastic stack (though it's pretty clear what `Pat's Scratch Box` is 😅)

this isn't a "fix this before we merge" comment; i'm very interested to hear your thoughts on this

Reply:
My only concern with not providing a default of the old value is what monitoring/automation others might have set up - doing it based on instance name isn't best practice but that doesn't mean it's not done, mostly taking guidance from https://github.com/buildkite/elastic-ci-stack-for-aws/issues/1028.  Happy to make it reference the StackName for the reasons you've given, also saves adding Parameter 😃

Reply:
oh man, contradicted by my past self 😅 on reflection i do think they're right though, we are indeed shackled by the past.

at some point in the medium term, we'll probably be releasing an elastic stack v6, that might be the right time to make the change, rather than shooting from the hip here.